File: Concurs Inot C#/RepoProba.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;
using log4net;

namespace Problema_3_CSharp.me.repository
{
    public class RepoProba : IRepoProba
    {
        private static readonly ILog log = LogManager.GetLogger("LoggerRepoProba");
        public Proba Delete(Proba elem)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Proba> FindAll()
        {
            log.Info("Se incepe conectarea la bd si selectarea tuturor probelor");
            var con = DBUtils.getConnection();
            IList<Proba> probe = new List<Proba>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Probe";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        Console.WriteLine("proba " + id);
                        float distanta = dataR.GetFloat(1);
                        Stiluri stil = (Stiluri)Enum.Parse(typeof(Stiluri), dataR.GetString(2));
                        Proba p = new Proba(id, distanta, stil);
                        probe.Add(p);
                    }
                }
            }
            return probe;
        }

        public IEnumerable<Proba> FindProbeDupaDistanta(float distanta)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Proba> FindProbeDupaStil(Stiluri stil)
        {
            throw new NotImplementedException();
        }

        public Proba Get(int id)
        {
            log.InfoFormat("Se cauta proba cu id-ul {0}", id);
            var con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Probe where id=@id";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        float distanta = dataR.GetFloat(1);
                        Stiluri stil = (Stiluri)Enum.Parse(typeof(Stiluri), dataR.GetString(2));
                        Proba p = new Proba(id, distanta, stil);
                        log.InfoFormat("S-a gasit {0}", p);
                        return p;
                    }
                }
            }
            log.InfoFormat("Nu s-a gasit nicio proba cu id-ul {0}", id);
            return null;
        }

        public Proba Save(Proba elem)
        {
            log.Info("Se incearca adaugarea");
            var con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {

                comm.CommandText = "insert into Probe (distanta,stil)  values (@distanta, @stil)";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@distanta";
                paramNume.Value = elem.Distanta;
                comm.Parameters.Add(paramNume);

                var paramVarsta = comm.CreateParameter();
                paramVarsta.ParameterName = "@stil";
                paramVarsta.Value = elem.Stil.ToString();
                comm.Parameters.Add(paramVarsta);

                var result = comm.ExecuteNonQuery();

                log.Info("S-a adaugat " + elem.ToString());
                return elem;
            }
        }

        public int Size()
        {
            throw new NotImplementedException();
        }

        public Proba Update(Proba elem)
        {
            throw new NotImplementedException();
        }
    }
}

File: RepoProba.java
package me.repository;

import me.model.Participant;
import me.model.Proba;
import me.model.ProbaParticipant;
import me.model.Stiluri;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoProba implements IRepoProba {
    private static final Logger logger = LogManager.getLogger();
    JdbcUtils jdbcUtils;

    public RepoProba(Properties props){
        logger.info("Initializing RepoAngajat with properties: {} ",props);
        jdbcUtils=new JdbcUtils(props);
    }

    @Override
    public List<Proba> findProbeDupaStil(Stiluri stil) { return null; }

    @Override
    public List<Proba> findProbeDupaDistanta(float distanta) { return null; }

    @Override
    public Proba save(Proba elem) {
        logger.info("Se conecteaza la baza de date");
        Connection con=jdbcUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into Probe (distanta,stil) values (?,?)")){
            preStmt.setFloat(1,elem.getDistanta());
            preStmt.setString(2,elem.getStil().toString());
            preStmt.executeUpdate();
        }catch (SQLException ex){
            logger.info("eroare baza de date: "+ex.getMessage());
        }
        logger.info("S-a adaugat in baza de date proba "+elem);
        return elem;
    }

    @Override
    public Proba update(Proba elem) { return null; }

    @Override
    public Proba delete(Proba elem) throws IllegalArgumentException {
        return null;
    }

    @Override
    public Proba get(Integer integer) {
        Connection con=jdbcUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Probe where id=?")){
            preStmt.setInt(1,integer);
            try(ResultSet result=preStmt.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    float distanta  = result.getFloat("distanta");
                    Stiluri stil = Stiluri.valueOf(result.getString("stil"));
                    return new Proba(id,distanta,stil);
                }
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }

    @Override
    public int size() { return 0; }

    @Override
    public Iterable<Proba> findAll() {
        logger.info("Se aduc toate probele...");
        Connection con=jdbcUtils.getConnection();
        List<Proba> probe = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Probe")){
            try(ResultSet result=preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("id");
                    float distanta = result.getFloat("distanta");
                    Stiluri stil = Stiluri.valueOf(result.getString("stil"));
                    probe.add(new Proba(id,distanta,stil));
                }
                return probe;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }
}
